### PR TITLE
Validate tool name format and add tests

### PR DESCRIPTION
**Pull request by kentcdodds**
Implement tool name validation according to the SEP to standardize naming and improve interoperability, issuing warnings for non-conforming names.

---
<a href="https://cursor.com/background-agent?bcId=bc-356e683f-b33f-41d0-9eb7-89efece561c2">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-cursor-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-cursor-light.svg">
    Open in Cursor
  </picture>
</a>
<a href="https://cursor.com/agents?id=bc-356e683f-b33f-41d0-9eb7-89efece561c2">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-web-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-web-light.svg">
    Open in Web
  </picture>
</a>

## Summary by CodeRabbit

* **New Features**
  * SEP-compliant tool name validation with user-facing warnings on register/rename.
  * Tools now support an optional _meta field surfaced in tool listings.
  * Server initialization and metadata types now accept website URL and icon descriptors.

* **Tests**
  * Comprehensive tests for name validation, warning messages, and edge cases.
  * Tests added for _meta behavior and improved test isolation with automatic mock cleanup.

### CODE REVIEW

File: src/server/mcp.test.ts [hunk 1]
```diff
@@ -1633,8 +1637,47 @@ describe("tool()", () => {
...
+    // Verify specific warning content
+    const warningCalls = warnSpy.mock.calls.map(call => call.join(' '));
+    expect(warningCalls.some(call => call.includes('Tool name starts or ends with a dash'))).toBe(true);
+    expect(warningCalls.some(call => call.includes('Tool name contains spaces'))).toBe(true);
+    expect(warningCalls.some(call => call.includes('Tool name contains invalid characters'))).toBe(true);
+  });
 });
 
+
+
```
Comment (kentcdodds) [inline src/server/mcp.test.ts]:
```suggestion
```

File: src/server/mcp.ts [hunk 2]
```diff
@@ -789,6 +793,10 @@ export class McpServer {
       remove: () => registeredTool.update({ name: null }),
       update: (updates) => {
         if (typeof updates.name !== "undefined" && updates.name !== name) {
+          // Validate new tool name if it's being changed
+          if (updates.name) {
+            validateAndWarnToolName(updates.name);
+          }
```
Comment (kentcdodds) [inline src/server/mcp.ts]:
```suggestion
          if (updates.name) validateAndWarnToolName(updates.name);
```

File: src/shared/toolNameValidation.test.ts [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+    expect(result.warnings).toHaveLength(0);
+  });
+
+  test('should handle names with mixed valid and invalid characters', () => {
+    const result = validateToolName('user@name123');
+    expect(result.isValid).toBe(false);
+    expect(result.warnings).toContain('Tool name contains invalid characters: "@"');
+  });
+
+
```
Comment (kentcdodds) [inline src/shared/toolNameValidation.test.ts]:
```suggestion
```

File: src/server/mcp.ts [hunk 4]
```diff
@@ -789,6 +793,7 @@ export class McpServer {
       remove: () => registeredTool.update({ name: null }),
       update: (updates) => {
         if (typeof updates.name !== "undefined" && updates.name !== name) {
+          if (updates.name) validateAndWarnToolName(updates.name);
```
Comment (kentcdodds) [inline src/server/mcp.ts]:
```suggestion
          if ('name' in updates) validateAndWarnToolName(updates.name);
```

File: src/server/mcp.ts [hunk 5]
```diff
@@ -789,6 +793,7 @@ export class McpServer {
       remove: () => registeredTool.update({ name: null }),
       update: (updates) => {
         if (typeof updates.name !== "undefined" && updates.name !== name) {
+          if ('name' in updates) validateAndWarnToolName(updates.name);
```
Comment (kentcdodds) [inline src/server/mcp.ts]:
```suggestion
          // don't use updates.name so we validate empty strings as well
          if ('name' in updates) validateAndWarnToolName(updates.name);
```
